### PR TITLE
godot-cpp: add version 4.1.1

### PR DESCRIPTION
**Pull request by fmorgner**
Specify library name and version:  **godot-cpp/4.1.1**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/godot-cpp/config.yml [hunk 1]
```diff
@@ -5,3 +5,5 @@ versions:
     folder: 3.x.x
   "cci.3.2-20200130":
     folder: 3.x.x
+  "4.1.1":
+    folder: all
```
Comment (Croydon) [inline recipes/godot-cpp/config.yml:9]: I would name the folder `4.x.x`

`all` only makes sense if there are no further directories

File: recipes/godot-cpp/config.yml [hunk 2]
```diff
@@ -1,7 +1,9 @@
 versions:
   "3.5.1":
-    folder: all
+    folder: 3.x.x
   "3.3.4":
-    folder: all
+    folder: 3.x.x
   "cci.3.2-20200130":
+    folder: 3.x.x
+  "4.1.1":
```
Comment (valgur) [inline recipes/godot-cpp/config.yml:8]: Please place the newest version on top.

File: recipes/godot-cpp/all/conanfile.py [hunk 3]
```diff
@@ -12,131 +10,43 @@ class GodotCppConan(ConanFile):
...
+        platform = {
+            "Android": "android",
+            "Linux": "linux",
+            "Macos": "darwin",
+            "Windows": "windows",
+        }[self.settings.get_safe("os")]
+        build_type = self.settings.get_safe("build_type").lower()
+
+        if self.settings.get_safe("os") == "Android":
+            self.cpp_info.libs= [f"godot-cpp.{platform}.{build_type}"]
```
Comment (valgur) [inline recipes/godot-cpp/all/conanfile.py:43]:
```suggestion
        }[self.settings.os]
        build_type = self.settings.build_type.lower()

        if self.settings.os == "Android":
            self.cpp_info.libs= [f"godot-cpp.{platform}.{build_type}"]
```

Comment (valgur) [reply recipes/godot-cpp/all/conanfile.py:43]: `build_type`, `os` and `arch` are standard settings that are guaranteed to be available.

File: recipes/godot-cpp/all/conanfile.py [hunk 4]
```diff
@@ -12,131 +10,43 @@ class GodotCppConan(ConanFile):
...
         else:
-            self.cpp_info.libs = [self._libname]
+            bits = 64 if self.settings.get_safe("arch") in ["x86_64", "armv8"] else 32
```
Comment (valgur) [inline recipes/godot-cpp/all/conanfile.py:45]:
```suggestion
            bits = 64 if self.settings.arch in ["x86_64", "armv8"] else 32
```

File: recipes/godot-cpp/config.yml (hunk 1 above)
Comment (valgur) [reply recipes/godot-cpp/config.yml:9]: I agree. Would also make the PR diff cleaner.

File: recipes/godot-cpp/all/conanfile.py (hunk 3 above)
Comment (fmorgner) [reply recipes/godot-cpp/all/conanfile.py]: adjusted by using `str(...)` where a string-like or hashable object is required.

**Comment by fmorgner** [comment]
I will look into the V2 build/lint of the old versions on the weekend.

**Comment by fmorgner** [comment]
I fixed the linter issues in 3.x.x (and one in 4.x.x). I tried to keep the changes small (although i did fix the format string warnings as well), since this PR is primarily focused on adding 4.1.1. I think that modernizing the 3.x.x conanfile should be done separately in a future PR (or by rebasing and adjusting #18862).

**Comment by fmorgner** [comment]
Is the v2 build expected to be re-triggered automatically (like the v1 one) or is there something I (or someone else) has to do?

**Review by valgur — APPROVED** [review]

**Comment by Croydon** [comment]
Can we pls get a CI re-run here? @uilianries @RubenRBS 

ConanV2 was in maintenance when it did run

**Comment by uilianries** [comment]
@Croydon Danke! I just re-triggered now.

**Comment by Croydon** [comment]
No new CI feedback?

**Comment by uilianries** [comment]
I just fixed some git conflicts, it triggers a new CI build that should post the result now.

**Pull request closed without merging**